conformance/lang/expressions/checking-expression/check_expr.balt

```diff
@@ -0,0 +1,2020 @@
+
+    string s1  = check rec["y"];
+    io:println(s1);
+
+    boolean[] b1 = check rec["z"];
+    io:println(b1);
+
+    anydata a1 = check rec["a"];
+    io:println(b1);
+
+    int i3 = check j1[0];
+    io:println(i3);
+
+    json _ = check j1[4].a;
+}
+
+Test-Case: error
+Description: Test the static type of the check-expr when it is used with field-access, optional-field-access and
+             member-access expressions. ***
+Labels: boolean, decimal, DecimalFloatingPointNumber, DecimalNumber, field-access-expr, float,
+        FloatingPointTypeSuffix, HexFloatingPointLiteral, HexIntLiteral, json, list-constructor-expr,
+        mapping-constructor-expr, nil-literal, readonly-type, string
+
+function init() {
+    error? x1 = file(); // @error expected '(int|error)' but found 'File'
+}
+
+class File {
+    public int i = 10;
+}
+
+function file() returns error? {
+    int f = check new File();
+}
+
+Test-Case: output
+Description: Test check-expr with function call expressions.
+Labels: boolean, DecimalNumber, error-type, function-defn, function-call-expr, optional-type, check, return-stmt, if-else-stmt
+        error-constructor-expr, string
+
+function init() {
+    error? x1 = testFunction1(); // @output function
+    io:println(x1); // @output error("error1!")
+
+    error? x2 = testFunction2();
+    io:println(x2); // @output error("error2")
+
+    error? x3 = testFunction3(5); // @output 5
+                                  // @output 4
+                                  // @output 3
+                                  // @output 2
+                                  // @output 1
+                                  // @output 0
+                                  // @output function
+    io:println(x3); // @output error("error1!")
+
+    error? x4 = testFunction3(2); // @output 2
+                                  // @output 1
+                                  // @output 0
+                                  // @output function
+    io:println(x4); // @output error("error1!")
+}
+
+function testFunction1() returns error? {
+    check fn();
+}
+
+function fn() returns error {
+    io:println("function");
+    return error("error1!");
+}
+
+function testFunction2() returns error? {
+    check fn1();
+}
+
+function fn1() returns error? {
+    check fn2();
+}
+
+function fn2() returns error? {
+    _ = check fn3() + check fn4(2);
+}
+
+function fn3() returns error|int {
+    return check fn4(1);
+}
+
+function fn4(int i) returns error|int {
+    if i > 1 {
+        return error("error2");
+    }
+    return 2;
+}
+
+function testFunction3(int i) returns error? {
+    check recursiveFn(i);
+}
+
+function recursiveFn(int i) returns error? {
+    io:println(i);
+    if i == 0 {
+        check fn();
+    }
+    check recursiveFn(i - 1);
+}
+
+Test-Case: output
+Description: Test check-expr with method call expressions.
+Labels: boolean, DecimalNumber, nil-literal, string, error-type, module-class-defn, explicit-new-expr, check, returm-stmt,
```

Comment:
```suggestion
Labels: boolean, DecimalNumber, nil-literal, string, error-type, module-class-defn, explicit-new-expr, check, 
```

Reply:
We have used `return-stmt` in this test. Ex. in `method2`, `return error("mothod2")`.

Reply:
Sorry, meant to change the spelling.

Reply:
My bad. I didn't notice that. Thanks.

Reply:
Will update it.